R/type-data-frame.R

```diff
@@ -94,6 +101,48 @@ df_ptype2 <- function(x, y, ..., x_arg = "", y_arg = "") {
   .Call(vctrs_df_ptype2, x, y, x_arg, y_arg)
 }
 
+vec_ptype2_df_fallback_normalise <- function(x, y) {
```

Comment:
@hadley could you check that the use of `[` in this function and the one below makes sense please? I am trying to insert the result of df-ptype2 and df-cast in the inputs, while letting class implementations restore attributes. The normalisation of columns must happen before the fallbacks.

Reply:
Merging now but please let me know if this looks fishy.